### ISSUE TITLE
Solve all pydoctor errors

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8565]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)| @adiroiban reported|
|-|-|
|Trac ID|trac#8565|
|Type|enhancement|
|Created|2016-07-05 17:55:15Z|
|Branch|https://github.com/twisted/twisted/tree/8565-no-pydoctor-errors|

this is the final ticket with which we should remove all pydoctor errors for the api-docs builder

<details><summary>Searchable metadata</summary>

```
trac-id__8565 8565
type__enhancement enhancement
reporter__adiroiban adiroiban
priority__normal normal
milestone__None None
branch__8565_no_pydoctor_errors 8565-no-pydoctor-errors
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1467741315947325 1467741315947325
changetime__1467855237583204 1467855237583204
version__None None
owner__adiroiban adiroiban

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_1"></a>|@adiroiban commented|
|-|-|

I have pushed an initial branch... will look at updating the tests.

The win32api is based on a hack in TwistedSphinxInventory ... but I don't like and I think that the whole TwistedSphinxInventory custom code should be moved to pydoctor

but I think that we can have it in Twisted for a while and after we find the best patter, we can move it in pydoctor

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban removed owner|
|-|-|

I have pushed an initial version.

This is a dirty hack as the win32 api is not related to Sphinx inventories...
and also the whole TwistedSphinxInventory should just be moved to pydoctor as normal SphinxInventory as other project might find this useful

Still, since Twisted is the major user of these features and I am not aware of any issue reported in pydoctor requesting this functionality I think that we should have this code hosted in Twisted for a while.

We can update it as we found more issues with inter API docs linking and when the fix is stable it can be moved to pydoctor.

If we will move this to pydoctor starting from now, each change will required a fix in pydoctor and then releasing a new pydoctor version.

This looks like a lot of work and people might be discouraged from providing cross references.

Please check the attached  branch.

With this branch api-documentation step should produce zero errors.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_3"></a>|@rodrigc set owner to @adiroiban|
|-|-|

In defer.py, just delete this sentence completely: **It is used only by the deprecated L{Deferred.setTimeout} method.**
Deferred.setTimeout() was removed in https://github.com/twisted/twisted/commit/0d2674cb7567e33ae8bdfca67ab9e4847643696e a while back.

Do that, then merge.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban removed owner|
|-|-|

There are 2 others tickets related to that docstring so at this point I just wanted to get rid of the errors. Whether we should remove or how to update that docstring should be left for one of those tickets.

In this branch I only care about bad pydoctor references.

Please also provide feedback for the  TwistedSphinxInventory hack.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_5"></a>|@rodrigc set owner to @adiroiban|
|-|-|

Easiest thing is to delete that sentence, but if you don't want to do that,
then remove that sentence from your patch.

TwistedSphinxInventory hack is not ideal but seems OK.

So either:
  (1)  Delete the sentence
  (2)  Leave the sentence out of your patch

Do one of those things, then merge.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_6"></a>|@adiroiban commented|
|-|-|

Thanks for the review.

I went for removing the sentence since the content is wrong and will follow up in 8533 to see how to write a proper docstring.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_7"></a>|@adiroiban set status to `closed`|
|-|-|

In changeset 556f0f24df2eba2f38ec7f0fa422c4aa7df07fec
```
#!CommitTicketReference repository="" revision="556f0f24df2eba2f38ec7f0fa422c4aa7df07fec"
Merge 8565-no-pydoctor-errors: Solve all pydoctor errors.

Author: adiroiban
Reviewer: rodrigc
Fixes: #8565
```
